src/analyses/raceAnalysis.ml

```diff
@@ -369,7 +369,7 @@ struct
   let special ctx (lvalOpt: lval option) (f:varinfo) (arglist:exp list) : D.t =
     (* perform shallow and deep invalidate according to Library descriptors *)
     let desc = LibraryFunctions.find f in
-    if List.mem LibraryDesc.ThreadUnsafe desc.attrs then (
+    if List.mem LibraryDesc.ThreadUnsafe desc.attrs && not (ctx.ask (Queries.MustBeSingleThreaded {since_start=true})) then (
```

Comment:
The same check as for normal accesses would be even better:
https://github.com/goblint/analyzer/blob/ea1bf23148b3babf04d4d2d844f4738b2c7e64e5/src/analyses/raceAnalysis.ml#L307
It also suppresses the accesses when the program has gone back single-threaded by joining other threads.